src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.cartitem.CartItemRequestDto;
+import com.example.bookstore.dto.shoppingcart.ShoppingCartDto;
+import com.example.bookstore.mapper.CartItemMapper;
+import com.example.bookstore.mapper.ShoppingCartMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.BookRepository;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.repository.UserRepository;
+import com.example.bookstore.service.ShoppingCartService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCartDto(String email) {
+        ShoppingCart shoppingCart = getShoppingCart(email);
+        shoppingCart.setCartItems(cartItemRepository.getAllByShoppingCart(shoppingCart));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addItemsToCart(String email, CartItemRequestDto requestDto) {
```

Comment:
we have a single item in the method param so lets name it accordingly
```suggestion
    @Override
    public ShoppingCartDto addItemToCart(String email, CartItemRequestDto requestDto) {
```